src/storage/mvcc/reader.rs

```diff
@@ -401,9 +406,158 @@ impl MvccReader {
+        assert!(self.lock_cursor.is_some());
+        assert!(self.write_cursor.is_some());
+
+        // Check lock.
+        match self.isolation_level {
+            IsolationLevel::SI => {
+                let lock = {
+                    let l_cur = self.lock_cursor.as_mut().unwrap();
+                    if l_cur.valid() && l_cur.key() == user_key.encoded().as_slice() {
+                        self.statistics.lock.processed += 1;
+                        Some(Lock::parse(l_cur.value())?)
+                    } else {
+                        None
+                    }
+                };
+                if let Some(l) = lock {
+                    self.check_lock_impl(user_key, ts, l)?;
+                }
+            }
+            IsolationLevel::RC => {}
+        }
+
+        // Get value for this user key.
+        let mut round = 0;
+        let mut lastest_version = (None /* start_ts */, None /* short value */);
+        let mut last_handled_key: Option<Vec<u8>> = None;
+        loop {
+            if !self.write_cursor.as_mut().unwrap().valid() {
+                return self.get_value(user_key, lastest_version.0, lastest_version.1);
+            }
+
+            // When there are many versions for this user key, after several tries,
+            // we will use seek to locate the right position. But this will turn around
+            // the write cf's iterator's direction inside RocksDB, and the next user key
+            // need to turn back the direction to backward. As we have tested, turn around
+            // iterator's direction from forward to backward is as expensive as seek in
+            // RocksDB, so don't set REVERSE_SEEK_BOUND too small.
+            if round >= REVERSE_SEEK_BOUND {
+                break;
+            }
+            round += 1;
+
+            let mut write = {
+                let (commit_ts, key) = {
+                    let w_cur = self.write_cursor.as_mut().unwrap();
+                    last_handled_key = Some(w_cur.key().to_vec());
+                    let w_key = Key::from_encoded(w_cur.key().to_vec());
+                    (w_key.decode_ts()?, w_key.truncate_ts()?)
+                };
+
+                // reach neighbour user key or can't see this version.
+                if ts < commit_ts || &key != user_key {
+                    assert!(&key <= user_key);
+                    return self.get_value(user_key, lastest_version.0, lastest_version.1);
+                }
+                self.statistics.write.processed += 1;
+                Write::parse(self.write_cursor.as_mut().unwrap().value())?
+            };
+
+            match write.write_type {
+                WriteType::Put => {
+                    if write.short_value.is_some() {
+                        if self.key_only {
+                            lastest_version = (None, Some(vec![]));
+                        } else {
+                            lastest_version = (None, write.short_value.take());
```

Comment:
I think we could define start_ts as u64 instead of an  Option, and load data if short_value is none.

Reply:
Short_value can be none when encounter DELETE.

Reply:
When encounter DELETE, short value and start ts both should be None.